### PR TITLE
Add top-level database_specific field

### PR DESCRIPTION
**Pull request by chrisbloom7**
This adds a top-level `database_specific` field to allow source databases to embed custom metadata that is relevant to the entire vulnerability rather than specific affected products.

### CODE REVIEW

**Review by oliverchang — APPROVED** [review]
Thanks! This looks pretty straightforward to me. I'll let others chime in here before merging.

**Comment by rsc** [comment]
I agree. This seems fine.

**Review by rsc — APPROVED** [review]

**Pull request merged**
